File: Linguagem de Programação I/Extras/Beecrowd_Extra/1025.java
import java.util.Arrays;
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int T = 1;
        while (true) {
            int N = scanner.nextInt();
            int Q = scanner.nextInt();
            if (N == 0 && Q == 0) break;

            int[] marmores = new int[N];
            for (int i = 0; i < N; ++i) {
                marmores[i] = scanner.nextInt();
            }

            Arrays.sort(marmores);

            System.out.println("CASE# " + T++ + ":");

            for (int i = 0; i < Q; ++i) {
                int consulta = scanner.nextInt();

                int index = Arrays.binarySearch(marmores, consulta);

                if (index < 0)
                    System.out.println(consulta + " not found");
                else {
                    // Arrays.binarySearch might not return the index of the first occurrence.
                    // We need to handle this case.
                    while (index > 0 && marmores[index - 1] == consulta) {
                        index--;
                    }
                    System.out.println(consulta + " found at " + (index + 1));
                }
            }
        }
    }
}

File: Linguagem de Programação I/Extras/Beecrowd_Extra/1025.go
package main

import (
    "fmt"
    "sort"
)

func main() {
    T := 1
    for {
        var N, Q int
        fmt.Scan(&N, &Q)
        if N == 0 && Q == 0 {
            break
        }

        marmores := make([]int, N)
        for i := 0; i < N; i++ {
            fmt.Scan(&marmores[i])
        }

        sort.Ints(marmores)

        fmt.Printf("CASE# %d:\n", T)
        T++

        for i := 0; i < Q; i++ {
            var consulta int
            fmt.Scan(&consulta)

            index := sort.SearchInts(marmores, consulta)

            if index < N && marmores[index] == consulta {
                fmt.Printf("%d found at %d\n", consulta, index+1)
            } else {
                fmt.Printf("%d not found\n", consulta)
            }
        }
    }
}